src/services/stepsService.ts

```diff
@@ -46,19 +52,19 @@ export class StepsService {
       } else {
         newStep.branches = newStep.branches.concat(branch);
       }
-      this.integrationJsonStore.replaceBranchParentStep(newStep, currentStepNested.pathToStep);
+      useIntegrationJsonStore.getState().replaceBranchParentStep(newStep, currentStepNested.pathToStep);
```

Comment:
This looks promising, as it still helps to avoid too many method arguments that could be retrieved from the stores - which also means the components have to dig into stores and then the business logic leaks into components.

Reply:
Actually it's better in a way that components don't have to pass the stores at all :+1: